src/holon/tests/test_rule_action_add.py

```diff
@@ -408,3 +408,67 @@ def test_rule_action_add_contract_multiple_and_remove_beforehand(self):
         cloned_contract_scope = updated_scenario.actor_set.get(original_id=contract_scope.id)
         assert added_contract_0.contractScope == cloned_contract_scope
         assert added_contract_1.contractScope == cloned_contract_scope
+
+    def test_rule_action_add_gridconnection_with_children(self):
+        """Test the add rule action for an actor with contracts"""
+
+        # Arrange
+        existing_contract_scope = Actor.objects.create(
+            category=ActorType.OPERATORGRID, payload=self.scenario
+        )
+        new_actor: Actor = Actor.objects.create(
+            category=ActorType.CONNECTIONOWNER,
+        )
+        DeliveryContract.objects.create(
+            name="template_delivery_contract",
+            energyCarrier=EnergyCarrier.ELECTRICITY,
+            actor=new_actor,
+            contractScope=existing_contract_scope,
+            deliveryContractType=DeliveryContractType.FIXED,
+            deliveryPrice_eurpkWh=1,
+            feedinPrice_eurpkWh=1,
+        )
+
+        gridconnection_0: BuildingGridConnection = BuildingGridConnection.objects.create(
+            owner_actor=new_actor,
+            capacity_kw=750.0,
+            # payload=self.scenario,
+            is_rule_action_template=True,
+            insulation_label=InsulationLabel.A,
+            heating_type=HeatingType.GASBURNER,
+            type=BuildingType.LOGISTICS,
+        )
+        ElectricHeatConversionAsset.objects.create(
+            gridconnection=gridconnection_0,
+            name="building_heat_pump",
+            type=ConversionAssetType.HEAT_PUMP_AIR,
+            eta_r=0.95,
+            deliveryTemp_degC=70.0,
+            capacityElectricity_kW=30.0,
+        )
+
+        rule = ScenarioRule.objects.create(
+            interactive_element_continuous_values=self.interactive_element_continuous_values,
+            model_type=ModelType.SCENARIO,
+        )
+        RuleActionAdd.objects.create(gridconnection_to_add=gridconnection_0, rule=rule)
+        interactive_elements = [{"value": "2", "interactive_element": self.interactive_element}]
+
+        # Act
+        updated_scenario = rule_mapping.get_scenario_and_apply_rules(
+            self.scenario.id, interactive_elements
+        )
+
+        # Assert
+        assert len(updated_scenario.gridconnection_set.all()) == 1  # was 0
```

Comment:
Shouldn't it be 2 since the value of the interactive element is 2?

Reply:
Voor RuleActionAdd wordt alleen aan modellen in je filtered queryset templates toegevoegd. Als je queryset kleiner is dan de interactief element waarde, dan wordt die waarde niet bereikt. In dit geval is je queryset alleen het scenario.
Voor het geval dat wel nodig is (dus 2 gridconnections toegevoegd), kan RuleActionAddMultiple gebruikt worden.